### ISSUE TITLE
mpileup without max-depth limitation

### ISSUE THREAD

**Issue by shalomhillelroth** [samtools/samtools#858]
Is it possible to enable running samtools mpileup without any max-depth?
We need such a feature to calculate exact mismatch rates.

Thanks!

**Comment by valeriuo** [comment]
Have you tried setting a very large value to max depth, e.g. `samtools mpileup -d  2147483647 reads.bam`?

**Comment by jkbonfield** [comment]
I tested setting max depth to 1 billion on a low depth sample and the memory usage didn't increase, which backs up my understanding of the algorith,.  The memory is *potential* memory growth, used if and when the depth gets high.  So I think it's safe, provided you do have the memory to handle the worst case scenario of genuine high depth.

Note that setting it too high may have overflow issues as the actual hard limit is max_depth parameter multiplied by number of samples (or files? needs checking).   We discussed this just now and I think it makes sense to have depth 0 as a way to indicate maximum, and get the code to then work out itself what maximum value that equates to.   (At present it won't work this way though so don't do that.)
